subworkflows/local/consensus/main.nf

```diff
@@ -76,18 +76,34 @@ workflow CONSENSUS {
         FASTQ_ALIGN_DNA(ch_reads_aligner_index_fasta, false)
         ch_versions = ch_versions.mix(FASTQ_ALIGN_DNA.out.versions)
 
-        def ch_mapped_bam = FASTQ_ALIGN_DNA.out.bam
+        def ch_mapped_bam = FASTQ_ALIGN_DNA.out.bam.map { meta, bam ->
+            def sample = meta.samplename ?: UUID.randomUUID().toString()
+            def new_bam = file("${sample}.mapped.bam")
+            bam.copyTo(new_bam)
+            tuple(meta, new_bam)
```

Comment:
This seems unnecessary, the samplename is a required value in the samplesheet so this will always be set. I'm also not sure what the reason is to copy the BAM file?